apps/designer/app/shared/db/assets.server.ts

```diff
@@ -15,11 +16,19 @@ export const loadByProject = async (projectId?: Project["id"]) => {
 
 export const create = async (
   projectId: Project["id"],
-  values: { type: string; name: string; path: string }
+  values: { name: string; path: string; size: number; arrayBuffer: ArrayBuffer }
 ) => {
+  // there is an issue in the @types/sharp, it also accepts array buffers
+  const image = sharp(values.arrayBuffer as Uint8Array);
+  const metadata = await image.metadata();
   const newAsset = await prisma.asset.create({
     data: {
-      ...values,
+      name: values.name,
+      path: values.path,
+      size: values.size,
+      format: metadata.format,
+      ...(metadata.width ? { width: Math.round(metadata.width) } : {}),
```

Comment:
should we round? idk

Reply:
yeah, it's incredibly rare that an image is like 1980.2 px but if it was we would break the db

Reply:
yeah, I wonder why an image can be floating point width and if we round, where will we have potential other issues, as an option we could have a db field type that supports float numbers

Reply:
So basically I changed the logic to always return a float so if the number is 10 it always returns 10.0 and the db type is float, this way it will never break